keywords/couchbaseserver.py

```diff
@@ -1182,16 +1183,47 @@ def rebalance_server(self, cluster_servers):
         self._wait_for_rebalance_complete()
         return True
 
+    def add_simple_document(self, bucket, scope, collection, doc_id, ipv6=False):
+        """Add a simple document to a collection"""
+        connection_url = choose_connection_url(self.cbs_ssl, ipv6, self.host)
+        sdk_client = get_cluster(connection_url, bucket, get_bucket=True)
+        collection_client = sdk_client.scope(scope).collection(collection)
 
-def get_sdk_client_with_bucket(ssl_enabled, cluster, cbs_ip, cbs_bucket):
-    if ssl_enabled and cluster.ipv6:
-        connection_url = "couchbases://{}?ssl=no_verify&ipv6=allow".format(cbs_ip)
-    elif ssl_enabled and not cluster.ipv6:
-        connection_url = "couchbases://{}?ssl=no_verify".format(cbs_ip)
-    elif not ssl_enabled and cluster.ipv6:
-        connection_url = "couchbase://{}?ipv6=allow".format(cbs_ip)
+        doc_body = doc_generators.simple()
+        doc_body["id"] = doc_id
+        try:
+            result = collection_client.insert(doc_id, doc_body)
+        except DocumentExistsException as e:
+            raise Exception("Tried to insert document that already exists: " + str(e)) from e
+        return result
+
+    def get_document(self, bucket, scope, collection, doc_id, ipv6=False):
+        """Retrieve a document from a collection"""
+        connection_url = choose_connection_url(self.cbs_ssl, ipv6, self.host)
+        sdk_client = get_cluster(connection_url, bucket, get_bucket=True)
+        collection_client = sdk_client.scope(scope).collection(collection)
+
+        try:
+            result = collection_client.get(doc_id)
+        except DocumentNotFoundException as e:
+            raise Exception("Tried to fetch document that does not exist: " + str(e)) from e
+        return result.content_as[dict]
+
+
+def choose_connection_url(ssl_enabled, ipv6, host):
```

Comment:
I feel like I've seen similar code to this somewhere, but can't find it. Was this copied from somewhere? Can we make some sort of sharing if that's the case?

Reply:
I can't find anything similar in the testkit either.